input/pagecontent/2-10-ContentSharing.md

```diff
@@ -42,16 +42,40 @@ Clients wishing to exchange structured information SHALL:
  
 If the `context.versionId` values match, the Hub proceeds with processing each of the FHIR resources in the Bundle and SHALL process all Bundle entries in an atomic manner.  After updating its copy of the current state of exchanged information, the Hub SHALL assign a new `context.versionId` to the anchor context and use this new `context.versionId` in the `[FHIR resource]-update` event it forwards to subscribed applications.  The Hub SHALL also include the `context.priorVersionId` in the distributed event which receiving applications MAY use to ensure they are apply the updates to the proper context version. The distributed update event SHALL contain a Bundle resource with the same Bundle `id` which was contained in the request.
 
-`[FHIR resource]-update` events MAY also update attributes of the anchor context or associated context resources.  For example, a Subscriber might wish to update the `status` attribute of the anchor context resource.
+`[FHIR resource]-update` events MAY also update attributes of the anchor context or associated context resources.  For example, a Subscriber might wish to update the `status` attribute of the anchor context resource.  If any updates are made during the content sharing session to a resource in the `context` array that was present in the `[FHIR resource]-open` event, the updated resources SHALL be present in the `Bundle` resource in the `updates` key.
 
 When a `[FHIR resource]-update` event is received by a Subscriber, the application should respond as is appropriate for its clinical use.  For example, an image reading application may choose to ignore an observation describing a patient's blood pressure.  Since transactional change sets are used during information exchange, no problems are caused by applications deciding to ignore exchanged information not relevant to their function.  However, they should read and retain the `context.versionId` of the anchor context provided in the event for later use.
 
+Resources referenced in a content sharing session SHOULD be accessible.  For example, a `ServiceRequest` referenced in an `ImagingStudy` should either be present in the content or a fullUrl SHOULD be provided.
+
+#### Context Array in Update and Select Events
+
+In addition to the `Bundle` resource in the `updates` key, the context array of `[FHIR resource]-update` and `[FHIR resource]-select` events SHALL include the only the anchor context resource which was present in the context array of the corresponding `[FHIR resource]-open` event.  The `Patient` resource which was present in the context array of the corresponding `[FHIR resource]-open` event MAY be present in `[FHIR resource]-update` events to enable Subscribers to mitigate patient safety concerns.
```

Comment:
Is this SHALL a requirement on implementers or authors of event specifications? The implementers SHALL include required elements in the context, right?

If it's a SHALL on spec authors, do we really need it? If so, can we consolidate requirements for spec authors away from requirements for implementers?

Reply:
This is a SHALL on both. Spec authors of new events and users of events following this pattern not explicitly defined in the event library